oak_functions/wasm/src/lib.rs

```diff
@@ -499,44 +452,60 @@ where
         })
     }
 
-    fn init_wasm_state(&self, request_bytes: Vec<u8>) -> anyhow::Result<WasmState<L>> {
+    // Create an extension from every factory in the WasmHandler.
+    fn create_extensions(
+        &self,
+    ) -> anyhow::Result<HashMap<ExtensionHandle, Box<dyn OakApiNativeExtension>>> {
         let mut extensions = HashMap::new();
-
-        // Create an extension from every factory.
         for factory in self.extension_factories.iter() {
             let extension = factory.create()?;
             extensions.insert(extension.get_handle(), extension);
         }
-
-        let wasm_state = WasmState::new(
-            self.wasm_module.clone(),
-            request_bytes,
-            self.logger.clone(),
-            extensions,
-        )?;
-
-        Ok(wasm_state)
+        Ok(extensions)
     }
 
-    pub fn handle_invoke(&self, request: Request) -> anyhow::Result<Response> {
-        let response_bytes = self.handle_raw_invoke(request.body)?;
-        Ok(Response::create(StatusCode::Success, response_bytes))
-    }
+    /// Handles a call to invoke by getting the raw request bytes from the body of the request to
+    /// invoke and returns a reponse to invoke setting the raw bytes in the body of the response.
+    pub fn handle_invoke(&self, invoke_request: Request) -> anyhow::Result<Response> {
+        let module = self.wasm_module.clone();
 
-    /// Handles an invocation using raw bytes and returns the response as raw bytes.
-    pub fn handle_raw_invoke(&self, request_bytes: Vec<u8>) -> anyhow::Result<Vec<u8>> {
-        let mut wasm_state = self.init_wasm_state(request_bytes)?;
+        let user_state = UserState::new(
+            invoke_request.body,
+            self.create_extensions()?,
+            self.logger.clone(),
+        );
+        // For isolated requests we need to create a new store for every request.
```

Comment:
I don't think we support non-isolated requests?

Reply:
Yes, that comment is a breadcrumb for ourselves to find how we guarantee isolated requests.